app/version.go

```diff
@@ -1,40 +1,89 @@
 package app
 
-import "github.com/urfave/cli/v2"
+import (
+	"fmt"
+	"runtime/debug"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/urfave/cli/v2"
+	"golang.org/x/mod/semver"
+)
 
 const (
-	DefaultVersion = "v0.13.0"
+	pseudoVersionMinLen        = len("vX.0.0-yyyymmddhhmmss-abcdefabcdef")
+	pseudoVersionCommitInfoLen = len("yyyymmddhhmmss-abcdefabcdef")
 )
 
 var (
-	BuildDate string
-	Commit    string
-	Version   string
+	MakeVersion string
 )
 
+type BuildInfo struct {
+	Commit     string
+	CommitTime time.Time
+	Version    string
+}
+
 func NewVersionCommand() (CommandOut, error) {
 	return CommandOut{Command: &cli.Command{
 		Name:    "version",
 		Usage:   "Version information",
 		Aliases: []string{"v"},
 		Action: func(c *cli.Context) error {
-			return PrintObj(&struct {
-				BuildDate string
-				Commit    string
-				Version   string
-			}{
-				BuildDate: BuildDate,
-				Commit:    Commit,
-				Version:   getVersion(),
-			})
+			return PrintObj(NewBuildInfo())
 		},
 	}}, nil
 }
 
-func getVersion() string {
-	version := Version
-	if len(version) == 0 {
-		version = DefaultVersion
+func NewBuildInfo() BuildInfo {
+	var info BuildInfo
+
+	di, ok := debug.ReadBuildInfo()
+	if !ok {
+		return info
 	}
-	return version
+
+	if semver.IsValid(di.Main.Version) {
+		info.Version = strings.Split(semver.Canonical(di.Main.Version), "-")[0]
```

Comment:
hmm should we leave the version as is instead of parsing out only the first part?

Reply:
Yeah that does sound easier too